src/Microsoft.Azure.SignalR.Common/Endpoints/AadAccessKey.cs

```diff
@@ -65,13 +62,14 @@ public Task<string> GenerateAadToken()
             return await base.GenerateAccessToken(audience, claims, lifetime, algorithm);
         }
 
-        public Task UpdateAccessKeyAsync(IServerNameProvider provider, ILoggerFactory loggerFactory)
+        public async Task UpdateAccessKeyAsync(IServerNameProvider provider, ILoggerFactory factory)
```

Comment:
Retryable failure seems acceptable and not need to log every time. Directly throw exception outside or make this `TryUpdateAccessKeyAsync()` to return true/false result to `AccessKeySynchronizer` to handle. And no need to pass in ILoggerFactory then.